sdk/core/src/request.rs

```diff
@@ -99,6 +99,10 @@ impl Request {
         &self.body
     }
 
+    pub fn set_headers(&mut self, headers: impl Into<Headers>) {
+        self.headers = headers.into();
+    }
+
```

Comment:
I'm somewhat worried that this just runs the risk of folks overriding any headers that have already been set. Perhaps an `append_headers` would be better? I wish this just worked with the existing `insert_headers`.

Reply:
I updated it to use `insert_headers`.